app/backend/src/couchers/models/users.py

```diff
@@ -399,11 +399,25 @@ class User(Base):
 
     @hybrid_property
     def has_completed_profile(self) -> bool:
-        return self.avatar_key is not None and self.about_me is not None and len(self.about_me) >= 150
+        # Check if user has at least one photo in their profile gallery
+        has_photo = self.profile_gallery is not None and len(self.profile_gallery.photos) > 0
+        return has_photo and self.about_me is not None and len(self.about_me) >= 150
 
     @has_completed_profile.expression
     def has_completed_profile(cls):
-        return (cls.avatar_key != None) & (func.character_length(cls.about_me) >= 150)
+        # Check if user has at least one photo in their profile gallery
+        from couchers.models import PhotoGalleryItem
```

Comment:
No inline imports please

Reply:
@aapeliv This is to avoid a circular import:

* User has profile_gallery_id → PhotoGallery
* PhotoGallery has owner_user_id → User

This creates a circular dependency at the module level, which Python can't resolve at import time.

Would you rather me move all gallery-checking logic to a separate service layer? We don't have a galleries service yet.